packages/editor/src/dataviews/store/private-actions.ts

```diff
@@ -28,3 +43,56 @@ export function unregisterEntityAction(
 		actionId,
 	};
 }
+
+export function setIsReady( kind: string, name: string ) {
+	return {
+		type: 'SET_IS_READY' as const,
+		kind,
+		name,
+	};
+}
+
+export const registerPostTypeActions =
+	( postType: string ) =>
+	async ( { registry }: { registry: any } ) => {
+		const isReady = unlock( registry.select( editorStore ) ).isEntityReady(
+			'postType',
+			postType
+		);
+		if ( isReady ) {
+			return;
+		}
+
+		unlock( registry.dispatch( editorStore ) ).setIsReady(
+			'postType',
+			postType
+		);
+
+		const postTypeConfig = ( await registry
+			.resolveSelect( coreStore )
+			.getPostType( postType ) ) as PostType;
+
+		const actions = [
+			postTypeConfig.slug === 'wp_block' ? exportPattern : undefined,
+			resetPost,
+			restorePost,
+			deletePost,
+			trashPost,
+			permanentlyDeletePost,
+		];
+
+		registry.batch( () => {
+			actions.forEach( ( action ) => {
+				if ( action === undefined ) {
+					return;
+				}
+				unlock( registry.dispatch( editorStore ) ).registerEntityAction(
+					'postType',
+					postType,
+					action
+				);
+			} );
+		} );
+
+		doAction( 'core.registerPostTypeActions', Promise.resolve(), postType );
```

Comment:
Could our action registration use the same hook?

Reply:
I can see two sides here:

 - Core code is not special and should use the same thing.
 - Adds complexity: figuring out when to register the action, which package and also no special treatment for performance...